src/resize.cc

```diff
@@ -840,6 +869,17 @@ class ResizeWorker : public NanAsyncWorker {
       image = composited;
     }
 
+    // Reverse premultiplication after all transformations:
+    if (shouldPremultiplyAlpha) {
+      VipsImage *imageUnpremultiplied;
+      if (Unpremultiply(hook, image, &imageUnpremultiplied)) {
+        (baton->err).append("Failed to unpremultiply alpha channel.");
+        return Error();
+      }
+
```

Comment:
Adding `vips_object_local(hook, imageUnpremultiplied);` here should remove the reference/memory leak.